src/energyplus/Test/ScheduleInterval_GTest.cpp

```diff
@@ -277,6 +277,113 @@ TEST_F(EnergyPlusFixture, ForwardTranslator_ScheduleFixedInterval_Hourly_Shifted
   EXPECT_EQ(8760, numUntils);
 }
 
+TEST_F(EnergyPlusFixture, ForwardTranslator_ScheduleFixedInterval_Days) {
+  // timestep 1 day
+  {
+    Model model;
+    Date start_date(MonthOfYear::Jan, 1, 2007);
+    Time timestep_day(1, 0);
+    Vector daily_wh_inlet_temperatures_c(5);
+    daily_wh_inlet_temperatures_c[0] = 10.0422222222222;
+    daily_wh_inlet_temperatures_c[1] = 9.98111111111111;
+    daily_wh_inlet_temperatures_c[2] = 9.92111111111111;
+    daily_wh_inlet_temperatures_c[3] = 9.86222222222222;
+    daily_wh_inlet_temperatures_c[4] = 9.80444444444445;
+    TimeSeries time_series_tmains(start_date, timestep_day, daily_wh_inlet_temperatures_c, "C");
+
+    boost::optional<ScheduleInterval> scheduleInterval = ScheduleInterval::fromTimeSeries(time_series_tmains, model);
+    ASSERT_TRUE(scheduleInterval);
+    EXPECT_TRUE(scheduleInterval->optionalCast<ScheduleFixedInterval>());
+    ScheduleFixedInterval scheduleFixedInterval = scheduleInterval->optionalCast<ScheduleFixedInterval>().get();
+    EXPECT_EQ(1440, scheduleFixedInterval.intervalLength());  // one day in minutes
+    EXPECT_EQ(1, scheduleFixedInterval.startMonth());
+    EXPECT_EQ(2, scheduleFixedInterval.startDay());  // unrelated to this test, but this should be fixed to be 1
+
+    // Forward translate the schedule
+    ForwardTranslator ft;
+    Workspace workspace = ft.translateModel(model);
+
+    std::vector<WorkspaceObject> objects = workspace.getObjectsByType(IddObjectType::Schedule_Compact);
+    ASSERT_EQ(1u, objects.size());
+
+    EXPECT_EQ("Through: 01/02",
+              objects[0].getString(2, false).get());  // this aligns with start day, but start day should be 1 (so this should be 01/01)
+    EXPECT_EQ("For: AllDays", objects[0].getString(3, false).get());
+    EXPECT_EQ("Until: 24:00", objects[0].getString(4, false).get());
+    EXPECT_EQ("10.0422222222222", objects[0].getString(5, false).get());
```

Comment:
This first "days" test ensures the first timestep translated is the first from model (and not the second timestep).